amity.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import print_function
+from room import Room
+from occupant import Occupant
+from occupant_gen import Occupants
+
+
+class Amity(object):
+
+    '''
+    A class to model the Amity building which
+    contains the various living and office spaces
+    '''
+    office_rooms = []
+    living_rooms = []
+    boarding_fellows = []
+    non_boarding_fellows = []
+    staff = []
+    allocations = []
+
+    def __init__(self):
+        # populate with model with rooms
+        self.populate()
+        # populate model with occupants
+        self.create_occupants()
+        # allocate the rooms to the potential occupants
+        self.allocate()
+
+    def create_occupants(self):
+        '''
+        Reads occupant details from txt file
+        and store them as occupant objects
+        '''
+        # read occupant's info from the occupants.txt file
+        with open('data/occupants.txt', 'r') as doc:
+            for line in doc.readlines():
+                line = line.strip()
+                if line:
+                    details = line.split()
+                    occupant1 = Occupant(name=details[0] + ' ' + details[1],
+                                         job_type=details[2])
+                    if details[2] == 'FELLOW' and details[-1] == 'Y':
+                        self.boarding_fellows.append(occupant1)
+                    elif details[-1] == 'N':
+                        self.non_boarding_fellows.append(occupant1)
+                    self.staff.append(occupant1)
+
+    def allocate(self):
+        '''
+        Calls the methods to randomly allocate
+        rooms to people
+        '''
+        # give staff offices that are closer together
+        available_staff_offices = self.get_available_rooms('OFFICE')
+        self.allocate_rooms(available_staff_offices, self.staff)
+        # give fellows offices that are closer together
+        available_fellow_offices = self.get_available_rooms('OFFICE')
+        self.allocate_rooms(available_fellow_offices,
+                            self.boarding_fellows + self.non_boarding_fellows)
+
+        # allocate rooms to boarding fellows
+        available_living_rooms = self.get_available_rooms('LIVING')
+        self.allocate_rooms(available_living_rooms, self.boarding_fellows)
+
+    def allocate_rooms(self, rooms, people):
+        '''
+        Allocates rooms to the list of people
+        '''
+        next_person = Occupants.next(people)
+        person = None
+        for room in rooms:
+            while room.is_not_full():
+                try:
+                    person = next_person.next()
+                    room.add_occupant(person)
+                except StopIteration:
+                    break
+
```

Comment:
**Code Quality: Critical**

You need to add a second level check to ensure when there are no more people, the iteration through rooms stops.